src/Raven.Server/ServerWide/Commands/Indexes/SetIndexStateCommand.cs

```diff
@@ -32,9 +32,14 @@ public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
             if (record.AutoIndexes.TryGetValue(IndexName, out AutoIndexDefinition autoIndex))
             {
                 autoIndex.State = State;
+                return null;
             }
-
-            return null;
+            if (record.Indexes.TryGetValue(IndexName, out IndexDefinition indexDefinition))
+            {
+                indexDefinition.State = State;
+                return null;
+            }
+            throw new InvalidOperationException($"Can not change index: {IndexName} state. Index doesn't exist");
```

Comment:
Do we have a test for that? I'm afraid it might kill the state machine and we will unable to perform further cluster operations.

Reply:
It kills the state machine. removing it